sound/soc/sof/intel/hda.c

```diff
@@ -409,6 +409,9 @@ static int hda_init_caps(struct snd_sof_dev *sdev)
 			return ret;
 	}
 
+	//FIXME: set code_mask to zero to detect codec again
+	bus->codec_mask = 0;
```

Comment:
@RanderWang just to confirm, we detect codecs at step 1 and HDMI at step 2 ? So to me I would replace the //FIXME comments in the code above and also add your commit message explanation as the comments stating what is detected where. This makes it obvious to any reader why the bus is reset twice etc.

Reply:
sorry @RanderWang , I mean you have a large comment with 3 steps, each comment step should be next to the line of code that does that step and not all in one place.